lux/lib/lux/lenses/discord/channel_lens.ex

```diff
@@ -0,0 +1,138 @@
+  Requires a Discord bot token in the application configuration:
+  ```elixir
+  config :lux, :discord_token, "your-bot-token"
+  ```
+
+  ## Rate Limiting
+  Discord API has rate limits per route. This lens handles rate limiting by:
+  - Respecting the retry_after header
+  - Following the global rate limit
+  - Using proper request headers
+
+  ## Examples
+      # Get a specific channel
+      iex> Lux.Lenses.Discord.ChannelLens.focus(%{
+      ...>   channel_id: "987654321"
+      ...> }, %{})
+      {:ok, %{channel: %{"id" => "987654321", "name" => "example-channel", ...}}}
+
+      # List channels in a guild
+      iex> Lux.Lenses.Discord.ChannelLens.focus(%{
+      ...>   guild_id: "123456789"
+      ...> }, %{})
+      {:ok, %{channels: [%{"id" => "987654321", "name" => "general", ...}]}}
+
+  ## Error Cases
+      # Invalid channel ID
+      iex> Lux.Lenses.Discord.ChannelLens.focus(%{channel_id: "invalid"}, %{})
+      {:error, %{type: "discord_api_error", code: 50004, message: "Invalid channel ID"}}
+
+      # Missing permissions
+      iex> Lux.Lenses.Discord.ChannelLens.focus(%{guild_id: "123456789"}, %{})
+      {:error, %{type: "discord_api_error", code: 50001, message: "Missing Access"}}
+
+      # Rate limited
+      iex> Lux.Lenses.Discord.ChannelLens.focus(%{channel_id: "123456789"}, %{})
+      {:error, %{type: "rate_limit", retry_after: 5000}}
+  """
+
+  use Lux.Lens,
+    name: "Discord Channel Lens",
+    description: "Reads Discord channel information",
+    url: "https://discord.com/api/v10",  # Base URL, actual endpoints handled in before_focus
+    method: :get,
+    headers: [{"content-type", "application/json"}],
+    auth: %{
+      type: :custom,
+      auth_function: &__MODULE__.add_auth_header/1
+    },
+    schema: %{
+      type: :object,
+      properties: %{
+        guild_id: %{
+          type: :string,
+          description: "The ID of the guild to list channels from",
+          pattern: "^[0-9]{17,20}$"
+        },
+        channel_id: %{
+          type: :string,
+          description: "The ID of the specific channel to get information about",
+          pattern: "^[0-9]{17,20}$"
+        }
+      },
+      oneOf: [
+        %{required: ["guild_id"]},
+        %{required: ["channel_id"]}
+      ]
+    }
+
+  require Logger
+
+  @doc """
+  Adds Discord bot token to request headers.
+  """
+  def add_auth_header(lens) do
+    token = Application.fetch_env!(:lux, :discord_token)
+    %{lens | headers: lens.headers ++ [{"Authorization", "Bot #{token}"}]}
+  end
+
+  @doc """
+  Prepares the request URL based on the parameters.
+  If channel_id is provided, fetches a specific channel using GET /channels/{channel.id}.
+  Otherwise, lists all channels in the guild using GET /guilds/{guild.id}/channels.
+  """
+  def before_focus(params) do
+    base_url = "https://discord.com/api/v10"
+
+    url = if Map.has_key?(params, :channel_id) do
+      "#{base_url}/channels/#{params.channel_id}"
```

Comment:
Wouldn't it be confusing to have one lenses loading either one channel or a list of channels? Why not have two separate lenses for this?